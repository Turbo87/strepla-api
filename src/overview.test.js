const fs = require('fs');
const { parseOverviewPage } = require('./overview');

describe('overview', () => {
  const TESTS = ['ec17', 'lev17'];

  for (let fixtureName of TESTS) {
    test(fixtureName, () => {
      let body = fs.readFileSync(`${__dirname}/../__fixtures__/${fixtureName}-overview.html`);
      expect(parseOverviewPage(body)).toMatchSnapshot();
    });
  }
});
