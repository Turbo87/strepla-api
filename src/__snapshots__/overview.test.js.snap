// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview ec17 1`] = `
Object {
  "classes": Array [
    "Hasen",
    "Igel",
  ],
  "days": Array [
    Object {
      "date": "2017-06-01",
      "results": Object {
        "Hasen": Object {
          "idDay": "5680",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=388&idDay=5680",
        },
        "Igel": Object {
          "idDay": "5698",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=388&idDay=5698",
        },
      },
    },
    Object {
      "date": "2017-06-02",
      "results": Object {
        "Hasen": Object {
          "idDay": "5701",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=388&idDay=5701",
        },
        "Igel": Object {
          "idDay": "5710",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=388&idDay=5710",
        },
      },
    },
    Object {
      "date": "2017-06-03",
      "results": Object {
        "Hasen": Object {
          "idDay": null,
          "link": null,
        },
        "Igel": Object {
          "idDay": null,
          "link": null,
        },
      },
    },
    Object {
      "date": "2017-06-04",
      "results": Object {
        "Hasen": Object {
          "idDay": null,
          "link": null,
        },
        "Igel": Object {
          "idDay": null,
          "link": null,
        },
      },
    },
    Object {
      "date": "2017-06-05",
      "results": Object {
        "Hasen": Object {
          "idDay": "5794",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=388&idDay=5794",
        },
        "Igel": Object {
          "idDay": "5797",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=388&idDay=5797",
        },
      },
    },
  ],
  "description": "Aachen Merzbr√ºck (EDKA) 01.06.2017-05.06.2017",
  "logoUrl": "http://strepla.de/scs/data/comp_388/data/logo.png",
  "name": "17. EuregioCup 2017",
}
`;

exports[`overview lev17 1`] = `
Object {
  "classes": Array [
    "Club",
  ],
  "days": Array [
    Object {
      "date": "2017-07-12",
      "results": Object {
        "Club": Object {
          "idDay": "6115",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=397&idDay=6115",
        },
      },
    },
    Object {
      "date": "2017-07-13",
      "results": Object {
        "Club": Object {
          "idDay": "6118",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=397&idDay=6118",
        },
      },
    },
    Object {
      "date": "2017-07-14",
      "results": Object {
        "Club": Object {
          "idDay": "6139",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=397&idDay=6139",
        },
      },
    },
    Object {
      "date": "2017-07-15",
      "results": Object {
        "Club": Object {
          "idDay": "6148",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=397&idDay=6148",
        },
      },
    },
    Object {
      "date": "2017-07-16",
      "results": Object {
        "Club": Object {
          "idDay": null,
          "link": null,
        },
      },
    },
    Object {
      "date": "2017-07-17",
      "results": Object {
        "Club": Object {
          "idDay": "6160",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=397&idDay=6160",
        },
      },
    },
    Object {
      "date": "2017-07-18",
      "results": Object {
        "Club": Object {
          "idDay": "6175",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=397&idDay=6175",
        },
      },
    },
    Object {
      "date": "2017-07-19",
      "results": Object {
        "Club": Object {
          "idDay": null,
          "link": null,
        },
      },
    },
    Object {
      "date": "2017-07-20",
      "results": Object {
        "Club": Object {
          "idDay": null,
          "link": null,
        },
      },
    },
    Object {
      "date": "2017-07-21",
      "results": Object {
        "Club": Object {
          "idDay": "6205",
          "link": "http://strepla.de/scs/Public/scoreDay.aspx?cId=397&idDay=6205",
        },
      },
    },
    Object {
      "date": "2017-07-22",
      "results": Object {
        "Club": Object {
          "idDay": null,
          "link": null,
        },
      },
    },
  ],
  "description": "Leverkusen Flugplatz (EDKL) 15.07.2017-22.07.2017",
  "logoUrl": "http://strepla.de/scs/data/comp_397/data/logo.png",
  "name": "Junioren-Quali Leverkusen",
}
`;
